test/jdk/java/util/Properties/PropertiesStoreTest.java

```diff
@@ -91,6 +98,36 @@
         };
     }
 
+    /**
+     * Returns a {@link Locale} to use for testing
+     */
+    @DataProvider(name = "localeProvider")
+    private Object[][] provideLocales() {
+        // pick a non-english locale for testing
+        Locale nonEnglishLocale = null;
+        for (Locale locale : Locale.getAvailableLocales()) {
+            // skip ROOT locale and ENGLISH language ones
+            if (!locale.getLanguage().isEmpty() && !locale.getLanguage().equals("en")) {
+                nonEnglishLocale = locale;
+                System.out.println("Selected non-english locale: " + nonEnglishLocale + " for tests");
+                break;
+            }
+        }
+        if (nonEnglishLocale == null) {
```

Comment:
Alternatively, create a Set<Locale>, to avoid duplicates, adding the candidates as they are discovered
and finally convert the Set to an Object[][]. It may be a bit easier to maintain.
```
private static Object[][] provideLocales() {
        // pick a non-english locale for testing
        Set<Locale> locales = Arrays.stream(Locale.getAvailableLocales())
                .filter(l -> !l.getLanguage().isEmpty() && !l.getLanguage().equals("en"))
                .limit(1)
                .collect(Collectors.toSet());
        locales.add(Locale.getDefault());
        locales.add(Locale.US);
        locales.add(Locale.ROOT);

        return locales.stream()
                .map(m -> new Locale[] {m})
                .toArray(n -> new Object[n][0]);
    }
```

Reply:
Thank you Roger for this suggestion. This indeed is cleaner. I've updated the PR to use this suggested code. The tests continue to pass with the latest round of changes.